src/prefect/context.py

```diff
@@ -534,7 +534,9 @@ def from_task_and_inputs(
             downstream_assets=task.assets[:]
             if isinstance(task, MaterializingTask) and task.assets
             else [],
-            upstream_assets=upstream_assets,
+            upstream_assets=list(
+                {asset.key: asset for asset in upstream_assets}.values()
+            ),
```

Comment:
Why is this different from the original?

Reply:
This was actually to get rid of duplicate assets, but I'm going to just convert all of these sets